scrapy_zyte_api/handler.py

```diff
@@ -100,12 +95,16 @@ def __init__(
         crawler.signals.connect(self.engine_started, signal=signals.engine_started)
         self._crawler = crawler
         self._fallback_handler = None
+        self._trust_env = settings.getbool("ZYTE_API_USE_ENV_PROXY")
 
     @classmethod
     def from_crawler(cls, crawler):
         return cls(crawler.settings, crawler)
 
-    def engine_started(self):
+    async def engine_started(self):
+        self._session = await self._client.session(
+            trust_env=self._trust_env
+        ).__aenter__()
```

Comment:
I'm not sure `__aenter__` is needed here. aiohttp.ClientSession example don't always use it (so, it seems not using a context manager is ok), and `__aenter__` is no-op, which just returns self.

Reply:
From https://docs.aiohttp.org/en/stable/client_quickstart.html:

> A session context manager usage is not mandatory but await session.close() method should be called in this case, e.g.:

Reply:
Hm, maybe it would also mean that there is no need for async def engine_started?

Reply:
> Hm, maybe it would also mean that there is no need for async def engine_started?

We might want to do that anyway nonetheless, to avoid [this warning](https://github.com/aio-libs/aiohttp/pull/1468).